### ISSUE TITLE
Install multiple Ubuntu

### ISSUE THREAD

**Issue by gonzela2006** [hakuna-m/wubiuefi#79]
Hello,

Currently, I have Windows 7 and I installed ubuntu 16.04 using wubiuefi. Can I also install Ubuntu 17.10 using wubiuefi so when my system boot, I can choose between Windows 7, Ubuntu 16.04 or Ubuntu 17.10.

Thanks

**Comment by hakuna-m** [comment]
For testing purpose, there is a branch for a [second install](https://github.com/hakuna-m/wubiuefi/tree/ubuntu2) with releases e.g. [1604r310-2](https://github.com/hakuna-m/wubiuefi/releases/tag/1604r310-2). Of course, we can upgrade this branch to 17.10.  But the drawback is that you need a new version for a third install, for a forth install and so on.

But no problem. If it is helpful for a lot of users, I'll upgrade the branch.

Additionally, there is also another workaround:  

- Rename the folder **ubuntu** of your current install to e.g. **ubuntu_1604**
- Install the new version
- The main version should be within **ubuntu** , other version should be in other folders. e.g. **ubuntu_1604**, **ubuntu_1710** and so on. 
- Add your install within **ubuntu_1604** to GRUB menu of the version within folder **ubuntu**  
`sudo nano /etc/grub.d/29_wubi`
  - Insert the following lines
```
#! /bin/sh -e

echo "Adding boot menu entry for Wubi"  >&2

cat << EOF
menuentry 'Wubi Ubuntu 16.04' {
search -s -f -n /ubuntu_1604/disks/root.disk
probe --set=diskuuid -u \$root
if loopback loop1 /ubuntu_1604/disks/root.disk; then
   set root='(loop1)'
   linux  /vmlinuz root=UUID=\$diskuuid loop=/ubuntu_1604/disks/root.disk ro quiet splash
   initrd /initrd.img
fi
}
menuentry 'Wubi Ubuntu 16.04 menu' {
search -s -f -n /ubuntu_1604/disks/root.disk
if loopback loop1 /ubuntu_1604/disks/root.disk; then
   set root='(loop1)'
   set prefix=(\$root)/boot/grub
   configfile (\$root)/boot/grub/grub.cfg
fi
}
EOF
```
  - Enable **/etc/grub.d/29_wubi**
```
sudo chmod +x /etc/grub.d/29_wubi
  - Update GRUB within folder **ubuntu**
sudo update_grub
```
- Update Ubuntu 16.04 within folder **ubuntu_1604**
  - Reboot with 'Wubi Ubuntu 16.04' 
  - Update GRUB
`sudo update-grub`
  - Replace **ubuntu** with **ubuntu_1604** in **/etc/fstab**
`sudo nano /etc/fstab`

**Comment by hakuna-m** [comment]
I upgraded the branch for a 2nd install and I built a new release: [1710r327-2](https://github.com/hakuna-m/wubiuefi/releases/tag/1710r327-2).

**Comment by gonzela2006** [comment]
Thank you very very much. You are intelligent :)
I tried your first solution and it is working well. On the next time, I'll try the 2nd install new release.
Thanks again

**Comment by BassemN** [comment]
Hello,
Can I use `1710r327-2` to install a second installation for Ubuntu 18.04?
Thanks

**Comment by BassemN** [comment]
Hello @hakuna-m 
Can I use 1710r327-2 to install a second installation for Ubuntu 18.04?
Thanks

**Comment by hakuna-m** [comment]
Probably, you need a fix for #91 which has been released with r330.

In general, it is intended that you can also use an older Wubi version for a newer Ubuntu version with  [various distros/versions feature](https://github.com/hakuna-m/wubiuefi/wiki#various-distrosversions). But as we don't know all changes of future releases, it is sometimes not posssible.

I'll upgrade the 2nd intall branch for 18.04.

P.S. Sorry for my late reply.

**Comment by bluedogerino** [comment]
hello @hakuna-m , i always get an error when installing linux mint mate 19 beta, Error unknown filesystem and the only way to get back into windows is by fixing mbr from windows 10 install disk.
Do u have any suggestions?

**Comment by hakuna-m** [comment]
@bluedogerino As Linux Mint Mate 19 is based on Ubuntu 18.04, probably [1710r327-2](https://github.com/hakuna-m/wubiuefi/releases/tag/1710r327-2) does not work. As I wrote above 

> Probably, you need a fix for #91 which has been released with r330.

Meanwhile I upgraded the [ubuntu2 branch](https://github.com/hakuna-m/wubiuefi/tree/ubuntu2) to 1804r331-2. I also created an exe but I haven't released it because of missing time for tests.

But I try to release it soon.

**Comment by bluedogerino** [comment]
@hakuna-m, thank you , but I'm tried using both of ur stable and unstable 1804 builds, they both said the same thing

**Comment by hakuna-m** [comment]
@bluedogerino But if you tried Wubiuefi for 18.04 and it doesn't work, a second install version for 18.04 doesn't help. So it is another issue. Can you open a new issue?

**Comment by bluedogerino** [comment]
ok

**Comment by BassemN** [comment]
@hakuna-m Thanks allot :)
Currently, I'm running Ubuntu 16.04 for about 1.5 year now to also install Ubuntu 18.04.1, I just need to install 18041r332-2?

**Comment by hakuna-m** [comment]
Yeah. The goal of [ubuntu2 branch](https://github.com/hakuna-m/wubiuefi/tree/ubuntu2) is that you can install a second Wubiuefi installation without deleting the first one.

So, if Ubuntu 16.04 is within default folder ubuntu (or another renamed folder excluding ubuntu2) , you can install Ubuntu 18.04.1 within folder ubuntu2 with [18041r332-2](https://github.com/hakuna-m/wubiuefi/releases/tag/18041r332-2).

But there are not only different folders for installation, there are different boot loader files, different boot menu entries and different entries for Windows to uninstall Wubiuefi.

**Comment by BassemN** [comment]
The last question please, Can I use 18041r332-2 to install Linux Mint instead of Ubuntu 18.04.1?

**Comment by hakuna-m** [comment]
In general, it is possible to install Ubuntu based versions of Linux Mint with [various distros/versions feature](https://github.com/hakuna-m/wubiuefi/wiki#various-distrosversions) (desktop environment **Wubi MD5**). But if you plan to install **Linux Mint 19 Tara**, there are possible GRUB issues (see #109).

**Comment by BassemN** [comment]
Thanks allot :+1: :+1:

**Comment by a000385** [comment]
@hakuna-m would you please write this (as tutorial) in the wiki section?
I think this is an important thing to know for me or other people who want to migrate to the new version but still can't delete the one currently in use.

Is this also working on wubiuefi for 14.04 or 16.04?
Maybe you could consider to update that 2 versions as those are LTS versions.

Thank you.

**Comment by hakuna-m** [comment]
> would you please write this (as tutorial) in the wiki section?
> think this is an important thing to know for me or other people who want to migrate to the new version but still can't delete the one currently in use.

Currently, features of testing branches like [ubuntu2](https://github.com/hakuna-m/wubiuefi/tree/ubuntu2) are not part of the wiki, but part of the release notes. e.g. [here]( https://github.com/hakuna-m/wubiuefi/releases/tag/18041r332-2).

The reason is that features should be only temporary in testing branches, sometimes only for one release.

But you are right, if it is not a temporary solution, we should add it to the wiki.  

> Is this also working on wubiuefi for 14.04 or 16.04?
> Maybe you could consider to update that 2 versions as those are LTS versions.

There is a two years old version for 16.04: [1604r310-2](https://github.com/hakuna-m/wubiuefi/releases/tag/1604r310-2)
But as [various distros/versions feature](https://github.com/hakuna-m/wubiuefi/wiki#various-distrosversions) allows to install older versions, there is IMHO no need to backport a testing branch. Upgrades should be only  sometimes necessary for newer versions.

**Comment by lemonyzer** [comment]
hey @hakuna-m ,
big thanks for this wonderfull tool and support!

will you release a second install build for Wubi for Ubuntu 20.04.1 (rev. 342)?

Thanks in advance!
lemon

**Comment by furllmm** [comment]
@hakuna-m  pls make windows tutorial for multiple installations... thx
